delaymodel.py

```diff
@@ -229,18 +236,46 @@ def calculate_delay(self, publish : bool = True):
         """
         while True:
             redis_publish_service_pulse(self.redis_obj, SERVICE_NAME)
-            loadtimes_on_hand = list(self.self.source_pointings_dict.keys())
-            if self.source_pointing_update.is_set():
-                #A new pointing has just arrived. Its loadtime must be checked and if within the 
-                #threshold, delay coefficients must be calculated for it and sent to the F-Engines.
-                new_loadtimes = list(set(loadtimes_on_hand) ^ set(list(self.self.source_pointings_dict.keys())))
-                new_loadtimes_sub_threshold = [loadtime for loadtime in new_loadtimes if loadtime <= self.threshold and loadtime > 0]
-
-
-            else:    
-                t = np.floor(time.time())
-                tts = [3, (TIME_INTERPOLATION_LENGTH/2) + 3, TIME_INTERPOLATION_LENGTH + 3] # Interpolate time samples with 3s advance
-                tts = np.array(tts) + t
+
+            #Now, we want to check whether the source pointing has updated OR whether it has been over 5 seconds since
+            #we last calculated delay coefficients:
+            t = time.time()
+            if ((t - self.delay_data["loadtime"]) >= 5 or self.source_point_update.is_set()):
+                #clear event here to avoid risk that the listening process sets the event as loadtime becomes > 5 and we miss a message.
+                self.source_point_update.clear()
+                t_int = np.round(t)
+                loadtime_from_now = None if self.source_pointings_dict['loadtime'] is None else self.source_pointings_dict['loadtime'] - t 
+
+                if loadtime_from_now is None:
+                    #loadtime is None. Update source pointing and calculate coefficients and set loadtime to 
+                    #self.min_loadtime_offset from now.
+                    self.source = SkyCoord(self.source_pointings_dict['ra'], self.source_pointings_dict['dec'], unit='deg')
+                    time_to_load = t_int + self.min_loadtime_offset
+                    sideband = self.source_pointings_dict['sideband']
+                    sslo = self.source_pointings_dict['sslo']
+                
+                elif (loadtime_from_now < self.min_loadtime_offset
```

Comment:
The comment here is a bit misleading, as it suggests the loadtime is bad.
Actually, it might just be that the command was valid when issued but is now very old, such that `load_time_from_now` is negative. This is perfectly reasonable behaviour.
(I think what the code actually _does_ is fine)